scripts/build/tests.js

```diff
@@ -64,40 +63,28 @@ async function runConsoleTests(runJs, defaultReporter, runInParallel, watchMode,
         testTimeout = 400000;
     }
 
-    if (tests || runners || light || testTimeout || taskConfigsFolder || keepFailed || shards || shardId) {
-        writeTestConfigFile(tests, runners, light, taskConfigsFolder, workerCount, stackTraceLimit, testTimeout, keepFailed, shards, shardId);
-    }
-
     const colors = cmdLineOptions.colors;
-    const reporter = cmdLineOptions.reporter || defaultReporter;
+    const reporterOption = ["reporter=" + (cmdLineOptions.reporter || defaultReporter),
+                            ...(keepFailed ? ["keepFailed=true"] : [])];
+
+    writeTestConfigFile("scripts/failed-tests", reporterOption,

```

Comment:
We weren't using these options in the parallel test runner, so we should be sure that setting them doesn't cause problems with `gulp runtests-parallel`.

Reply:
It *looks* like it's working, in the sense that it produces almost the same output -- same number of tests, almost the same time, etc.  (But see general comment.)